File: Quizz_Models/DTO/PermissionDTO.cs
using System;

namespace Quizz_Models.DTO
{
    public class PermissionDTO
    {
        public int PkPermission { get; set; }
        public string Nom { get; set; }
        public Nullable<bool> GenererQuizz { get; set; }
        public Nullable<bool> AjouterQuest { get; set; }
        public Nullable<bool> ModifierQuest { get; set; }
        public Nullable<bool> SupprQuestion { get; set; }
        public Nullable<bool> ModifierCompte { get; set; }
        public Nullable<bool> SupprimerCompte { get; set; }
    }
}

File: Quizz_Web/ClientApp/src/app/DTO/PermissionDTO.ts
export class PermissionDTO {
    private Nom             : string;
    private GenererQuizz    : boolean;
    private AjouterQuest    : boolean;
    private ModifierQuest   : boolean;
    private ModifierCompte  : boolean;
    private SupprQuestion   : boolean;
    private SupprimerCompte : boolean;

    constructor(Nom: string, GenererQuizz: boolean, AjouterQuest: boolean, ModifierQuest : boolean, ModifierCompte : boolean,
        SupprQuestion : boolean, SupprimerCompte : boolean){
        this.Nom = Nom;
        this.GenererQuizz       = GenererQuizz;
        this.AjouterQuest       = AjouterQuest;
        this.ModifierQuest      = ModifierQuest;
        this.ModifierCompte     = ModifierCompte;
        this.SupprQuestion      = SupprQuestion;
        this.SupprimerCompte    = SupprimerCompte;
    }

    public get GetNom(): string{
        return this.Nom;
    }

    public get GetGenererQuizz(): boolean{
        return this.GenererQuizz;
    }

    public get GetAjouterQuest(): boolean{
        return this.AjouterQuest;
    }
    
    public get GetModifierQuest(): boolean{
        return this.ModifierQuest;
    }
    
    public get GetModifierCompte(): boolean{
        return this.ModifierCompte;
    }

    public get GetSupprQuestion(): boolean{
        return this.SupprQuestion;
    }

    public get GetSupprimerCompte(): boolean{
        return this.SupprimerCompte;
    }
}
